src/logic/startAccountMigration.js

```diff
@@ -0,0 +1,54 @@
+import {createLogic} from 'redux-logic';
+
+import {first} from 'rxjs/operators';
+
+import {migrateAccount} from '../clients/firebase';
+import {
+  accountMigrationComplete,
+  accountMigrationError,
+  accountMigrationUndoPeriodExpired,
+} from '../actions/user';
+import {getCurrentAccountMigration} from '../selectors';
+import {bugsnagClient} from '../util/bugsnag';
+
+export default createLogic({
+  type: 'START_ACCOUNT_MIGRATION',
+  async process({getState, action$}, dispatch, done) {
+    const continuePromise = new Promise((resolve) => {
+      setTimeout(resolve, 5000, false);
+    });
+
+    const cancelPromise = action$.pipe(first(
+      ({type}) => type === 'DISMISS_ACCOUNT_MIGRATION',
+      false,
```

Comment:
Closely linked to my [comment on the test above](https://github.com/popcodeorg/popcode/pull/1701/files#diff-37da696ec8de026c3ea562a8340aeb2bR51), I don’t think the optional default value makes sense here—`action$` is a continuous stream of actions that continues for the lifetime of the application. A default value for `first()` would only be needed if the observable might complete without ever emitting another value, which as discussed above is not something we would expect to happen in real-world code (and thus should not simulate in tests either).